tests/testsuite/publish.rs

```diff
@@ -158,23 +162,17 @@ fn duplicate_version() {
         .file("src/main.rs", "fn main() {}")
         .build();
 
+    p.cargo("publish")
+        .replace_crates_io(registry_dupl.index_url())
+        .without_status()
+        .run();
```

Comment:
This is publishing the package to ensure we get the error.

Does
```rust
    Package::new("too_new_child", "0.0.1")
        .rust_version("1.2345.0")
        .file("src/lib.rs", "fn other_stuff() {}")
        .publish();
```
work with the `RegistryBuilder` created earlier?

Reply:
This works with the fix, replacing the extra `cargo publish` call.